src/highdicom/pr/content.py

```diff
@@ -0,0 +1,634 @@
+
+from pydicom.dataset import Dataset
+
+import numpy as np
+
+from highdicom.color import CIELabColor
+from highdicom.content import LUT, ReferencedImageSequence, VOILUT
+from highdicom.enum import VOILUTFunctionValues
+from highdicom.pr.enum import (
+    AnnotationUnitsValues,
+    GraphicTypeValues,
+    TextJustificationValues,
+)
+from highdicom.uid import UID
+from highdicom.utils import is_tiled_image
+from highdicom.valuerep import (
+    _check_code_string,
+    _check_long_string,
+    _check_short_text
+)
+
+
+class GraphicLayer(Dataset):
+
+    """A layer of graphic annotations that should be rendered together."""
+
+    def __init__(
+        self,
+        layer_name: str,
+        order: int,
+        description: Optional[str] = None,
+        display_color: Optional[CIELabColor] = None
+    ):
+        """
+
+        Parameters
+        ----------
+        layer_name: str
+            Name for the layer.  Should be a valid DICOM Code String (CS), i.e.
+            16 characters or fewer containing only uppercase letters, spaces
+            and underscores.
+        order: int
+            Integer indicating the order in which this layer should be rendered.
+            Lower values are rendered first.
+        description: Union[str, None], optional
+            A description of the contents of this graphic layer.
+        display_color: Union[CIELabColor, None], optional
+            A default color value for rendering this layer.
+
+        """
+        super().__init__()
+        _check_code_string(layer_name)
+        self.GraphicLayer = layer_name
+        if not isinstance(order, int):
+            raise TypeError('"order" must be an integer.')
+        self.GraphicLayerOrder = order
+        if description is not None:
+            _check_long_string(description)
+            self.GraphicLayerDescription = description
+        if display_color is not None:
+            if not isinstance(display_color, CIELabColor):
+                raise TypeError(
+                    '"recommended_display_color" must be of type '
+                    'highdicom.color.CIELabColor.'
+                )
+            self.GraphicLayerRecommendedDisplayCIELabValue = list(
+                display_color.value
+            )
+
+
+class GraphicGroup(Dataset):
+
+    """Dataset describing a grouping of annotations.
+
+    Note
+    ----
+    Graphic Groups are an independent concept from Graphic Layers. Where a
+    Graphic Layer (:class:`highdicom.pr.GraphicLayer`) specifies which
+    annotations are rendered first, a Graphic Group specifies which annotations
+    belong together and shall be handled together (e.g., rotate, move)
+    independent of the Graphic Layer to which they are assigned.
+
+    Each annotation (:class:`highdicom.pr.GraphicObject` or
+    :class:`highdicom.pr.TextObject`) may optionally be assigned to a
+    GraphicGroup upon construction (whereas assignment to a
+    :class:`highdicom.pr.GraphicLayer` is required.)
```

Comment:
This is very helpful. While I understand the difference between layers and groups conceptually, I still struggle to come up with a practical example where a two objects may be in the same group but in different layers or vice versa. Do you have a practical example or use case that would illustrate the difference?

Reply:
Added a further explanation in 6387a70

Reply:
Nice! Thanks